### DISCUSSION TITLE
Delaunay

### DISCUSSION THREAD

**Discussion by HFenice** [PMEAL/OpenPNM#2021] [Q&A]
Hi,

I have a cylindrical sample and I need to reproduce its pore network. I used `DelaunayVoronoiDual` that let me create a cylinder, but I don't know if this choice is consistent with the reality. How can I choose between `Delaunay` and `DelaunayVoronoiDual`? Can I have a cylinder by simply using `Delaunay`? I need to compute permeability and I saw that there is a difference in the value obtained between using these two network.

**Comment by jgostick** [comment]
It depends what you're trying to do.  I'm guessing that your domain is a cylinder (like a tube) and it's filled with solid grains (sphere packing)?  In this case it is generally true that the sphere-to-sphere network (i.e. the solid network) resembles a delaunay tessellation while the pore space around the grains follows a voronoi tessellation ([see here](https://aiche.onlinelibrary.wiley.com/doi/abs/10.1002/aic.690390303)).  So this means you want a voronoi network.  The tricky part is figuring out which spatial points to provide to the tessellation, since these points should represent the sphere centers.  Anyway, assuming you have sphere centers, you would create a voronoi with ``pn = op.network.Voronoi(points=points, shape=[R])``.

**Reply by HFenice** [reply]
My domain is a cylinder but in my case instead of solid grains I have bubbles (macroporosity) connected by a matrix (microporosity). So I want that the pores of my network represent the bubbles and the throats represent the matrix. Does the DelaunayVoronoiDual network work for this?

**Reply by jgostick** [reply]
In this case then the 'pores' will be described by the Delaunay network rather than the Voronoi, which actually makes things much easier for you. If you know the bubble centers you can pass these in to the class initialization.  How much do you know about your bubbles/holes? Do you have any 3D images?  Do you know their size distribution?  How much overlap is there between bubbles?

**Reply by HFenice** [reply]
Yes, I have a tomography in Avizo but I can't generate the pore network with this software because tomography resolution doesn't see matrix microporosity but only the macroporosity of the bubbles, which are pores created by a foam. So my idea is to use the pore size distribution that I have in Avizo and to create a random network (it is not important bubbles position for me) where the pores are the bubbles and the throats represent the microporosity of my matrix.

**Reply by jgostick** [reply]
Hi @HFenice I think we should have a live chat.  We have been working on a tool that may work for you, but it's not published yet.  Can you connect with me via email?

**Comment by jgostick** [comment]
Your other option is to find or generate an image of spheres packed in a cylinder then just run the ``snow2`` network extraction on it, using ``porespy.networks.snow2``.

**Comment by HFenice** [comment]
Hi,

sorry I've just read the message. Yes, of course.
